### ISSUE TITLE
NullPointerException from 0.8.0

### ISSUE THREAD

**Issue by hamberluo** [socketio/socket.io-client-java#343]
NullPointerException from 0.8.0

```
Fatal Exception: java.lang.NullPointerException: Attempt to invoke virtual method 'io.socket.b.a io.socket.engineio.client.c.b(java.lang.String, io.socket.b.a$a)' on a null object reference
       at io.socket.engineio.client.Socket$7.call(SourceFile:346)
       at io.socket.emitter.Emitter$OnceListener.call(SourceFile:164)
       at io.socket.emitter.Emitter.emit(SourceFile:117)
       at io.socket.engineio.client.Transport.onOpen(SourceFile:122)
       at io.socket.engineio.client.transports.WebSocket.access$002(SourceFile:28)
       at io.socket.engineio.client.transports.WebSocket$2$1.run(SourceFile:93)
       at io.socket.thread.EventThread$2.run(SourceFile:80)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
       at java.lang.Thread.run(Thread.java:818)
```

**Comment by nkzawa** [comment]
How do you reproduce ?

**Comment by eckovation** [comment]
@nkzawa possibly when the phone is coming offline from online, and the socket gets connected, and lots of events are being received by mobile. But this replication is not sure-shot.

**Comment by eckovation** [comment]
lately, this has been causing constant crashes, in so many phones.

my worked around was to use an older library with Chendrak's fix.
compile 'com.github.chendrak:engine.io-client-java:engine.io-client-0.7.4'
        compile('io.socket:socket.io-client:0.7.0') {
            // excluding org.json which is provided by Android
            exclude group: 'org.json', module: 'json'
            exclude group: 'io.socket', module: 'engine.io-client'
        }

**Comment by IlyaDiallo** [comment]
> lately, this has been causing constant crashes, in so many phones.

Did you have the problem only since switching to 0.8.0 from 0.7.0 or did you have it before ...?

**Comment by eckovation** [comment]
@IlyaDiallo only after i upgraded to 0.8.0. But it seems, it has something to do the else. Because even after downgrading to 0.7.0 it hasn't stopped. Any possible cue on what might be the case?

**Comment by niranjanMeteor** [comment]
I am also facing the same Null Pointer Call Stack .Please help me out about its remedy

**Comment by eckovation** [comment]
It seems that, the @chendrak changes caused this error again! because sometimes, at the following place, the transport[0] comes out to be null.

"transport[0].once(Transport.EVENT_PACKET, new Listener() {"

This happens, most likely, src/main/java/io/socket/engineio/client/transports/WebSocket.java  in the write function. (the case could be that connection has died).

Also, this error ALWAYS occurs when polling to being upgraded to WebSocket.

**Comment by eckovation** [comment]
@nkzawa @chendrak I have found the sure shot way of replicating this bug.

Compiled this library into an Android app. Then place a your debugger on https://github.com/socketio/engine.io-client-java/blob/master/src/main/java/io/socket/engineio/client/Socket.java#L345
And the moment it reaches there, DISCONNECT YOUR INTERNET.
Then just continue the execution.
It will then crash at the very next line, and give you the EXACT same crash log, as we have mentioned above.
https://github.com/socketio/engine.io-client-java/blob/master/src/main/java/io/socket/engineio/client/Socket.java#L346

Kindly fix it as soon as possible.

**Comment by rajeefmk** [comment]
@nkzawa Referencing an old similar issue, in case it helps - https://github.com/socketio/socket.io-client-java/issues/127

**Comment by nkzawa** [comment]
Released v`0.8.1` !
